swan-lake/development-tutorials/deployment-guide/k8s-deployment.md

```diff
@@ -0,0 +1,354 @@
+```
+- Build the GraalVM executable in Kubernetes
+
+  GraalVM enhances Kubernetes deployments of Ballerina applications by enabling faster startup times and reduced memory usage. These advantages make it ideal for microservices workloads that demand rapid scaling and minimal resource consumption in a containerized environment. GraalVM’s native image feature uses ahead-of-time compilation, producing lightweight containers and improving Kubernetes deployment efficiency. The GraalVM build is best suited for deployments that prioritize fast scaling and reduced resource usage, while the standard JVM build is more appropriate for services that do not require such optimizations.
+
+Execute the command below to build a Ballerina package.
+```
+$ bal build --graalvm --cloud=k8s
+```
+See [Build the GraalVM executable in a container](https://ballerina.io/learn/build-the-executable-in-a-container/) for more information.
+
+7. Deploy the Kubernetes artifacts 
+```
+ $ kubectl apply -f target/kubernetes/hello_k8s
+```
+8. Testing the Kubernetes Deployment
+   Verifying the Kubernetes deployment ensures that the application is running correctly and meets the expected functionality within the Kubernetes environment.
+  - Verify the pods
+    ```
+    $ kubectl get pods
+    ```
+  - Expose the service via NodePort
+    ```
+    $ kubectl expose deployment hello-k8s-deployment --type=NodePort --name=hello-k8s-svc-local exposed
+    ```
+  - Get the External IP and port
+    ```
+    $ kubectl get svc
+    NAME                      TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)          AGE
+    hello-k8s-svc         ClusterIP   10.96.173.207   <none>        9090/TCP,9091/TCP               5m11s
+    hello-k8s-svc-local   NodePort    10.99.245.41    <none>        9090:30342/TCP,9091:30515/TCP   66s
+    kubernetes            ClusterIP   10.96.0.1       <none>        443/TCP                         130m
+    ```
+  - Execute the curl http://192.168.49.2:30342/helloWorld/greeting command to access the deployed service via cURL.
+    ```
+     $ curl http://192.168.49.2:30342/helloWorld/greeting
+     Hello, Kubernetes!
+    ```
+    Follow the below references for more information:
+  - [Deploy Ballerina on Kubernetes](https://ballerina.io/learn/deploy-ballerina-on-kubernetes/)
+  - [Kubernetes - Hello world](https://ballerina.io/learn/by-example/kubernetes-hello-world/)
+
+## Ballerina deployment with Kubernetes Kustomize
+
+[Kustomize](https://kustomize.io/) is a tool that allows you to modify Kubernetes YAML files without altering the original files. It can enhance the generated YAML from code-to-cloud deployments by applying additional customizations. The `kustomization.yaml` file in the root directory serves as a sample, combining the generated YAML files from different projects. Environment variables, such as specifying the location of the `Config.toml` for a service, can be added using Kustomize patches, enabling further configuration.
```

Comment:
```suggestion
[Kustomize](https://kustomize.io/) is a tool that allows you to modify Kubernetes YAML files without altering the original files. It can enhance the generated YAML from code-to-cloud deployments by applying additional customizations. The `kustomization.yaml` file in the root directory serves as a sample, combining the generated YAML files from different projects. Using Kustomize patches, you can add environment variables, such as specifying the location of the `Config.toml` file for a service, to enable additional configuration.
```

Reply:
Is it correct to say `root directory serves as a **sample**`? Why do we say `sample` here?